gson/src/test/java/com/google/gson/FieldNamingPolicyTest.java

```diff
@@ -0,0 +1,78 @@
+package com.google.gson;
+
+import org.junit.Test;
+import org.junit.runners.Parameterized.Parameters;
+
+import static org.junit.Assert.assertEquals;
+
+public class FieldNamingPolicyTest {
+	
+
+	@Test
+	public void testUpperCaseFirstLetterNonUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("someFieldName", "SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("SomeFieldName", "SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUnderscoreAndNonUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("_someFieldName", "_SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUnderScoreAndWithUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("_SomeFieldName", "_SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUnderscoreAndNumberAndNonUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("_999someFieldName", "_999SomeFieldName");
+	}
+	
+	@Test
+	public void testUpperCaseFirstLetterWithUnderScoreAndNumberWithUpperCaseLetter() {
+		verifyUpperCaseFirstLetter("_333SomeFieldName", "_333SomeFieldName");
+	}
+
+	@Test
+	public void testUpperCaseFirstLetter1LowercaseLetter() {
+		verifyUpperCaseFirstLetter("x", "X");
+	}
+
+	@Test
+	public void testUpperCaseFirstLetter1UppercaseLetter() {
+		verifyUpperCaseFirstLetter("X", "X");
+	}
+
+	@Test
+	public void testUpperCaseFirstAllNumbers() {
+		verifyUpperCaseFirstLetter("555", "555");
+	}
+
+	@Test
+	public void testUpperCaseFirstJustUnderscore() {
+		verifyUpperCaseFirstLetter("_", "_");
+	}
+
+	@Test
+	public void testUpperCaseBlank() {
+		verifyUpperCaseFirstLetter(" ", " ");
```

Comment:
I'll make parameterized tests for these, and commit those.  The reason I don't stack them all in one is because when I do put in some code which breaks the tests, I find it desirable to have the summary of what I broke, where as, if they're all in one method, once 1 assertion fails, the test executes.  So, did I do something that broke 10 of 15 scenarios, or just 1.  I'll get this in, in the next commit